src/Features/CSharp/Portable/ConvertAutoPropertyToFullProperty/CSharpConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            AccessorDeclarationSyntax getter = null;
+            AccessorDeclarationSyntax setter = null;
+            foreach (var accessor in accessorListSyntax.Accessors)
+            {
+                if (accessor.Kind() == SyntaxKind.GetAccessorDeclaration)
+                {
+                    getter = accessor;
+                }
+                else if (accessor.Kind() == SyntaxKind.SetAccessorDeclaration)
+                {
+                    setter = accessor;
+                }
+            }
+
+            return (getAccessor: getter, setAccessor: setter);
+        }
+
+        private static void GetExistingAccessors(AccessorListSyntax accessorListSyntax, ref AccessorDeclarationSyntax getAccessor, ref AccessorDeclarationSyntax setAccessor)
+        {
+            foreach (var accessor in accessorListSyntax.Accessors)
+            {
+                if (accessor.Kind() == SyntaxKind.GetAccessorDeclaration)
+                {
+                    getAccessor = accessor;
+                }
+                else if (accessor.Kind() == SyntaxKind.SetAccessorDeclaration)
+                {
+                    setAccessor = accessor;
+                }
+            }
+        }
+
+        private async Task<SyntaxNode> AddStatementsToAccessorAsync(
+                Document document,
+                SyntaxNode accessor,
+                SyntaxNode statement,
+                SyntaxGenerator generator,
+                CancellationToken cancellationToken)
+        {
+            var newAccessor = UpdateAccessor(accessor, statement);
+            newAccessor = await ConvertToExpressionBodyIfDesiredAsync(
+                document,
+                newAccessor,
+                cancellationToken).ConfigureAwait(false);
+
+            return await Formatter.FormatAsync(newAccessor, document.Project.Solution.Workspace).ConfigureAwait(false);
+        }
+
+        internal async Task<SyntaxNode> ConvertToExpressionBodyIfDesiredAsync(
+            Document document,
+            SyntaxNode accessor,
+            CancellationToken cancellationToken)
+        {
+            var accessorDeclarationSyntax = (AccessorDeclarationSyntax)accessor;
+
+            var preference = await GetExpressionBodyPreferenceAsync(document, cancellationToken).ConfigureAwait(false);
+            if (preference == ExpressionBodyPreference.Never)
+            {
+                return accessorDeclarationSyntax.WithSemicolonToken(default);
+            }
+
+            var ableToConvert = accessorDeclarationSyntax.Body.TryConvertToExpressionBody(
+                accessorDeclarationSyntax.Kind(),
+                accessor.SyntaxTree.Options,
+                preference,
+                out var arrowExpression,
+                out var semicolonToken);
+
+            // Should always be able to convert to expression body since we are creating the accessor 
+            // and know that it only has one statement
+            Debug.Assert(ableToConvert);
```

Comment:
This it not true. For example, you might be in C# 5.  We won't be able to conver to an expression body there.